fluss-rpc/src/main/java/com/alibaba/fluss/rpc/netty/client/ServerConnection.java

```diff
@@ -404,7 +417,16 @@ private void sendAuthenticate(byte[] challenge) {
 
     private void handleAuthenticateResponse(ApiMessage response, Throwable cause) {
         if (cause != null) {
-            close(cause);
+            if (cause instanceof RetriableAuthenticationException) {
+                LOG.warn("Authentication failed, retrying {} times", retryAuthCount, cause);
+                channel.eventLoop()
+                        .schedule(
+                                this::sendInitialToken,
+                                backoff.backoff(retryAuthCount++),
```

Comment:
We should add a big lock synchronized for the whole `handleAuthenticateResponse` method.